lib/Feature/FeatureModelTransaction.cpp

```diff
@@ -36,6 +39,98 @@ void removeFeature(FeatureModel &FM,
   Trans.commit();
 }
 
+Feature *getActualFeature(FeatureModel &FM, detail::FeatureVariantTy &FV) {
+  Feature *ActualFeature = nullptr;
+  std::visit(
+      Overloaded{[&ActualFeature, &FM](Feature *F) { ActualFeature = F; },
+                 [&ActualFeature, &FM](string &FName) {
+                   ActualFeature = FM.getFeature(FName);
+                 }},
+      FV);
+  return ActualFeature;
+}
+
+std::optional<bool> fvIsLeave(FeatureModel &FM, detail::FeatureVariantTy &FV) {
+  Feature *ActualFeature = getActualFeature(FM, FV);
+  // if Feature does not exist in FM
+  if (ActualFeature == nullptr) {
+    return std::nullopt;
+  }
+  return std::optional<bool>{ActualFeature->isLeave()};
+}
+
+std::optional<bool>
+canBeDeletedRecursively(FeatureModel &FM, detail::FeatureVariantTy &FV,
```

Comment:
Could you please write a quick doc string for this funciton.

Reply:
Done